apps/src/codebridge/MiniAppPreview/NeighborhoodPreview.tsx

```diff
@@ -0,0 +1,78 @@
+import {useCodebridgeContext} from '@codebridge/codebridgeContext';
+import {MiniApps} from '@codebridge/constants';
+import {appendSystemOutMessage} from '@codebridge/redux/consoleRedux';
+import React, {useEffect, useMemo} from 'react';
+
+import {setIsRunning} from '@cdo/apps/lab2/redux/systemRedux';
+import skins from '@cdo/apps/maze/skins';
+import Neighborhood from '@cdo/apps/miniApps/neighborhood/Neighborhood';
+import NeighborhoodVisualization from '@cdo/apps/miniApps/neighborhood/NeighborhoodVisualization';
+import {useAppDispatch, useAppSelector} from '@cdo/apps/util/reduxHooks';
+
+// Preview panel for the neighborhood mini app.
+const NeighborhoodPreview: React.FunctionComponent = () => {
+  const levelProperties = useAppSelector(state => state.lab.levelProperties);
+  const dispatch = useAppDispatch();
+  const {config} = useCodebridgeContext();
+  const isVertical = config.activeGridLayout === 'vertical';
+
+  const neighborhoodSkin = useMemo(() => {
+    if (!levelProperties) {
+      return null;
+    }
+    return skins.load(
+      (path: string) => levelProperties.baseAssetUrl + path,
+      MiniApps.Neighborhood
+    );
+  }, [levelProperties]);
+
+  useEffect(() => {
+    if (!levelProperties || !neighborhoodSkin) {
+      return;
+    }
+    const neighborhood = new Neighborhood(
+      message => dispatch(appendSystemOutMessage(message)),
+      () => dispatch(appendSystemOutMessage('')),
+      isRunning => dispatch(setIsRunning(isRunning)),
+      '[PYTHON LAB]'
+    );
+
+    neighborhood.afterInject(
+      levelProperties,
+      neighborhoodSkin,
+      {
+        skinId: MiniApps.Neighborhood,
+        level: levelProperties,
+        skin: neighborhoodSkin,
+      },
+      () => {},
+      () => {},
+      () => {},
+      () => {}
+    );
+
+    // The vertical version of the mini app is a static size for now,
+    // so we can hard-code the css. The horizontal version is resizable,
+    // and the css is handled by WorkspaceAndOutput.
+    if (isVertical) {
+      $('#visualization').css({
+        'max-width': '400px',
```

Comment:
Nit: do you need to set 'max-width' and 'max-height' here?

Reply:
Seems like no! I think I had needed it before I did some other css and didn't realize it was no longer necessary.